### PR TITLE
fix arithmetic mul operator make -0

### PR DESCRIPTION
**Pull request by laotan332**
for #2351

### CODE REVIEW

File: be/src/exprs/vectorized/arithmetic_operation.h [hunk 1]
```diff
@@ -61,6 +61,10 @@ struct ArithmeticBinaryOperator {
         } else if constexpr (is_sub_op<Op>) {
             return l - r;
         } else if constexpr (is_mul_op<Op>) {
+            // avoid 0 mul a negative num, make result -0
+            if (UNLIKELY(l == LType(0) || r == RType(0))) {
```
Comment (stdpain) [inline be/src/exprs/vectorized/arithmetic_operation.h:65]: Just add this judgment for floating point type

**Comment by dirtysalt** [comment]
Does `-fno-signed-zeros` work?

File: be/src/exprs/vectorized/arithmetic_operation.h [hunk 2]
```diff
@@ -62,14 +62,18 @@ struct ArithmeticBinaryOperator {
             return l - r;
         } else if constexpr (is_mul_op<Op>) {
             // avoid 0 mul a negative num, make result -0
-            if (UNLIKELY(l == LType(0) || r == RType(0))) {
-                return LType(0);
+            if constexpr (std::is_same<LType, double>::value || std::is_same<LType, float>::value) {
```
Comment (stdpain) [inline be/src/exprs/vectorized/arithmetic_operation.h:65]: prefer PrimitiveType == TYPE_DOUBLE instead of RuntimeCppType

**Comment by laotan332** [comment]
> 

it maybe cause other arithmetic operator slow

**Comment by laotan332** [comment]
run unittest

**Comment by laotan332** [comment]
run starrocks_be_unittest

**Review by kangkaisen — APPROVED** [review]

**Review by Youngwb — APPROVED** [review]

**Pull request merged**